### ISSUE TITLE
Dishonored [BLUS30501] lacks Vulkan and OpenGL Parity

### ISSUE THREAD

**Issue by Darkhost1999** [RPCS3/rpcs3#13929]
### Quick summary

Dishonored displays a black screen when using Vulkan
Dishonored works well enough when using OpenGL

### Details

On RPCS3 I have beaten the first mission after the escape from Prison mission using OpenGL + Accurate Gettllar.
0.0.22-13636 worked with Vulkan I just need to clear shaders cache every single time I boot the game and sometimes spu cache. 
0.0.22-13637 Vulkan + Dishonored just refuses to work all the way up to current master regardless of config.
But I really don't believe https://github.com/RPCS3/rpcs3/pull/12018 created the issue. Because the previous 5 to 10 builds was when the game was starting issues and problems with inconsistencies. Going even older makes Vulkan work freely. But why use Vulkan on an old build when OpenGL just works on a modern build.

### Attach a log file

[Vulkan.log.gz](https://github.com/RPCS3/rpcs3/files/11594268/RPCS3.log.gz)
[OpenGL.log.gz](https://github.com/RPCS3/rpcs3/files/11594282/RPCS3.log.gz)

### Attach capture files for visual issues

![image](https://github.com/RPCS3/rpcs3/assets/60384196/ebb9257b-b03a-4cfc-9ebf-145698791eb4)
Vulkan isn't letting me create an RSX capture
![image](https://github.com/RPCS3/rpcs3/assets/60384196/252ba2fc-2285-4548-a958-b3b4816dfe8e)
[BLUS30501_20230529164439_capture.zip](https://github.com/RPCS3/rpcs3/files/11594285/BLUS30501_20230529164439_capture.zip)

### System configuration

Ryzen 5 3600
GTX 1660 Super
16 Gigs of ram
WIndows 11 beta insider

### Other details

It's not everyday you find games that don't work with Vulkan that does work with OpenGL. Normally it's the other way around. This feels like one of those issues that's like the games that "still" don't work with llvm stuff. But with graphics instead.
I really wish I had more information and if there is more information that I can dig up from this stuff I would love to provide it.
I just boot my game watch the intro videos and before the Press Start screen is when Vulkan will just display a black screen so it's not very informative. OpenGL just works so IDK.

**Comment by kd-11** [comment]
One large difference between the 2 is that OpenGL executes work serially while vulkan is fully asynchronous. This means from the CELL point of view, vulkan runs very fast then stalls for a while to render everything before proceeding while opengl just moves through the command queue at a steady pace.
Try using atomic FIFO with vulkan. In theory that should be enough to force the renderer to slow down enough.

**Comment by Darkhost1999** [comment]
Debug output slows. it down enough 😃 
![image](https://github.com/RPCS3/rpcs3/assets/60384196/a99809d2-d260-4587-944c-ba7685ae0760)
[RPCS3.log.gz](https://github.com/RPCS3/rpcs3/files/11598056/RPCS3.log.gz)

**Comment by Darkhost1999** [comment]
Max driver wakeup delay + atomic fifo + high upscale values seems to be a workaround for now.
![image](https://github.com/RPCS3/rpcs3/assets/60384196/aa723be7-6d96-4318-9a9d-8af815e557f4)
[RPCS3.log.gz](https://github.com/RPCS3/rpcs3/files/11599248/RPCS3.log.gz)
Is it possible to improve synchronization at default scale and settings?

**Comment by digant73** [comment]
@Darkhost1999 is that still an issue?

**Comment by Darkhost1999** [comment]
Yes, Dishonored doesn't display anything whilst using Vulkan without slowing Vulkan down significantly.

**Comment by Darkhost1999** [comment]
Test of PR 18170
<img width="1920" height="1032" alt="Image" src="https://github.com/user-attachments/assets/0e2bdcb4-c7cb-4999-a30c-2fd2744cd5ab" />

[RPCS3.log](https://github.com/user-attachments/files/25213398/RPCS3.log)

It froze but at least it rendered something with vulkan.
Reason my test failed before was that my global config was set to 1080p to test and disprove someone's comment with Killzone 1 HD.

**Comment by Darkhost1999** [comment]
<img width="1920" height="1032" alt="Image" src="https://github.com/user-attachments/assets/7591f683-f100-4188-bd14-ea4f5e93e3ff" />

[RPCS3.log](https://github.com/user-attachments/files/25213462/RPCS3.log)

Current master for comparison either froze before drawing to the screen or just can't draw the title screen.

**Comment by Darkhost1999** [comment]
Current observations suggest PR 18170 may satisfy the intent of this issue ticket

**Comment by kd-11** [comment]
Also check if host GPU labels helps

**Comment by Darkhost1999** [comment]
There's almost no consistency in how far I'll get into the game before a freeze. Accurate gettllar was the old setting that almost eliminated freezes.
There's no obvious visual difference.
It's just nice vulkan is fully rendering the whole expected image and I can even get in game moving too now.

**Comment by Darkhost1999** [comment]
Current status is that we can't reuse shader cache else Dishonored will not render a title screen when using Vulkan. OGL is fine to reuse cache.
There are frequent freezes very likely unrelated to RSX stuff and possibly related to MFC things.
With clean cache gameplay can achieve ingame state and play for any unknown duration of time until things halt and you gotta restart the game. But you gotta clear shader cache first.

<img width="1920" height="1032" alt="Image" src="https://github.com/user-attachments/assets/66947e08-c053-4eb9-b54d-891b52410f73" />
Here is a moment where the shader cache was cleared AND when the game decided to freeze to stop making any progress

[RPCS3.log](https://github.com/user-attachments/files/25419608/RPCS3.log)

Then here is the nothing rendered no title screen attempt to reuse cache
[RPCS3.log](https://github.com/user-attachments/files/25419644/RPCS3.log)

**Comment by Darkhost1999** [comment]
Main menu issue appears to only present itself on consecutive boots. Clearing shader cache helps to make the menu work once. Then upon reboot get a black screen.
